Grunt build

Andertech opened the pull request:
This makes up for both assets managing and build/push to gh-pages. It should work ok AFAIK and tested. @mtrias can you help me reviewing this? (also I'm still missing the part when I check the branch calling the ```grunt publish``` command).

mtrias commented on prod_config.yaml:
```diff
@@ -0,0 +1,21 @@
+  env: "prod"
+  AuthorName: "VividCortex"
+  GitHubUser: "VividCortex"
+  GTMID: "KSGK44"
```
why is the GTMID duplicated in config.yaml ?

Andertech replied:
I got it removed already :)

mtrias commented on grunt/aliases.yml:
```diff
@@ -0,0 +1,18 @@
+  tasks:
+    - 'less:prod'
+    - 'shell:build'
+    - 'gh-pages:test'
```
why the "test" here ?

Andertech commented:
Ok, I added these checks for publishing: 

- developer should be in master to run the ```grunt publish``` task
- ```grunt publish``` will check there are no uncommited changes before building
- and it will also check with the remote master branch to see if either the local or the remote are out of sync.

Also fixed most of the issues you pointed out in the already closed navigation-fix PR and this one. Overall I think I would rather merge this branch now if there's nothing wrong with the grunt build, so I can focus on the smaller issues that keep me from going live. 

To actually run the tasks locally without running risks to deploy or otherwise make unnecessary changes I suggest you making another branch with this one as a start and changing this on line 14 of the Gruntfile:

```
if (currentBranch === 'master') { // If the current branch is master move unto the next task, and display an ok message ...
```
to the name of the branch you are currently working on, and do the same on the gitinfo task at line 4, by adding ```/name-of-your-testing-branch``` to the end of the ```'origin-SHA': ['rev-parse', '--verify', 'origin``` . 

Sorry this is a pain, I haven't thought of another way of testing these things out for them to actually work without the need to do these changes. 

I couldn't figure out why the styles are not getting rendered on vividcortex.github.io/docs any ideas @nachocaru ?

nachocaru commented:
@Andertech everything is inside the root dir so we should remove the first " / " in all the calls like "/img/filename.jpg" change to "img/filename.jpg"

mtrias commented:
code LGTM. :ship:

Andertech commented:
Thanks @nachocaru MIND == BLOWN
